### DISCUSSION TITLE
Filter does not work in Edge function

### DISCUSSION THREAD

**Discussion by deliqs** [supabase/supabase#11540] [Questions] [answered]
Hi, 

I'm having a weird issue that I can't fix. I have an Edge endpoint that my customers use to insert certain data into my database. At the start of this function, I query my database for their API client to determine which customer is inserting the data based on the provided API key. I do that via the following code:

```
const apiKey = req.headers.get('x-api-key');
if (!apiKey?.length) {
    const errorMessage = 'No API key found at header "x-api-key".';
    return sendErrorResponse(errorMessage, 400);
}

const { data: apiClients } = await supabaseAdmin.from<ApiClient[]>("apiclients")
    .select()
    .eq("apikey", apiKey);
```

I have debugged this code by checking the incoming API key and the response of the query. The API key is correct and unique in the table. Still, the query returns **all rows** from that table. Because the API keys are unique, you would expect that the query should return a single ApiClient, but it does not. The API Edge Network logs do show this as search path: 
`"search": "?select=*&apikey=eq.XXXXXXX"`. So it seems like the filter is sent to the database.

The supabaseAdmin is instantiated via this:
```
export const supabaseAdmin = createClient(
    Deno.env.get("SUPABASE_URL") ?? "",
    Deno.env.get("SUPABASE_SERVICE_ROLE_KEY") ?? ""
);
```

I have already tried to replace the `.eq()` with `.match({})`, but that has the same result.
The table has RLS enabled without any added policies. There are also no functions/triggers on the table. 

To be able to continue, I have found this workaround.
```
const response = await supabaseAdmin
    .from("apiclients")
    .select()
    .eq("apikey", apiKey)
    .throwOnError();

const apiClients = response.data.filter(x => x.apikey === apiKey);
```

Does anyone know what happens here?

**Comment by soedirgo** [comment, marked as answer]
`apikey` is a reserved column name [[1]](https://supabase.com/docs/reference/javascript/select) [[2]](https://github.com/supabase/supabase/issues/5465) - sorry this isn't prominent enough in the docs. Changing the column name to e.g. `api_key` should fix it.

**Reply by soedirgo** [reply]
We might lift this restriction at some point, though there's no timeline for it yet.

**Reply by deliqs** [reply]
Heheh, that explains a lot. Thanks @soedirgo!
